### PR TITLE
parsec: added build-time and smoke testing

### PR DESCRIPTION
**Pull request by abouteiller**
Invoke ctest -j1 from `spack install --test root parsec`, 
and build_with_parsec from `spack test run parsec`

Signed-off-by: Aurelien Bouteiller <bouteill@icl.utk.edu>

### CODE REVIEW

**Review by alalazo — APPROVED** [review]

**Review by alalazo — DISMISSED** [review]
Too fast approving :sweat_smile: Can you please fix style errors:
```
var/spack/repos/builtin/packages/parsec/package.py:72: [E303] too many blank lines (2)
var/spack/repos/builtin/packages/parsec/package.py:74: [E501] line too long (89 > 88 characters)
var/spack/repos/builtin/packages/parsec/package.py:75: [E501] line too long (107 > 88 characters)
```
?

File: var/spack/repos/builtin/packages/parsec/package.py [hunk 1]
```diff
@@ -53,3 +56,28 @@ def cmake_args(self):
             self.define_from_variant('PARSEC_DEBUG_PARANOID', 'debug_verbose'),
         ]
         return args
+
+    def check(self):
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/parsec/package.py:62]:
```suggestion
    @run_after('install')
    def check(self):
```

Comment (abouteiller) [reply var/spack/repos/builtin/packages/parsec/package.py:62]: This change causes the check method to run all the time after install, even when `--test root` is not passed as an argument to `spack install`.
These are quite a lot of tests, maybe they should not run all the time. I am misunderstanding something about how `--test` works?

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/parsec/package.py:62]: You are correct.  You would need to add the following between the `run_after` and `def check` lines if you only want the tests to be run when the `--test` option is specified on the command line:

```suggestion
    @on_package_attributes(run_tests=True)
    def check(self):
```

**Comment by alalazo** [comment]
@abouteiller Can you please rebase and push?

**Comment by tldahlgren** [comment]
Would you mind changing line 39 from:

```
    depends_on('cmake@3.16.0:', type='build')
```

to

```
    depends_on('cmake@3.16:', type='build')
```

(i.e., remove `.0` from `3.16.0`)?

I know it's not part of your PR but I'm working on a system with `CMake` at `3.16.8` installed (but not my default) and the above change allows me to run:

```
$ spack install parsec ^cmake@3.16
```

without a version range error.

**Comment by tldahlgren** [comment]
Just a heads up: I haven't quite figured out the problem but running with build-time tests seems to hang for me.

**Comment by tldahlgren** [comment]
Strange build test failure so closing and re-opening to force the PR checks to re-run.

**Comment by abouteiller** [comment]
It's not immediately clear to me that the reported failure is related to this PR

**Review by tldahlgren — APPROVED** [review]
LGTM

**Pull request merged**